src/ComputeSharp.D2D1/Shaders/Interop/D2D1PixelShader.cs

```diff
@@ -263,6 +263,46 @@ public static ReadOnlyMemory<byte> GetConstantBuffer<T>(in T shader)
         return dataLoader.GetResultingDispatchData();
     }
 
+    /// <summary>
+    /// Gets the constant buffer from an input D2D1 pixel shader.
+    /// </summary>
+    /// <typeparam name="T">The type of D2D1 pixel shader to retrieve info for.</typeparam>
+    /// <param name="shader">The input D2D1 pixel shader to retrieve info for.</param>
+    /// <param name="span">The target <see cref="Span{T}"/> to write the constant buffer to.</param>
+    /// <returns>The number of bytes written into <paramref name="span"/>.</returns>
+    /// <exception cref="ArgumentException">Thrown if <paramref name="span"/> is not large enough to contain the constant buffer.</exception>
+    public static int GetConstantBuffer<T>(in T shader, Span<byte> span)
+        where T : unmanaged, ID2D1PixelShader
+    {
+        if (!TryGetConstantBuffer(in shader, span, out int writtenBytes))
+        {
+            ThrowHelper.ThrowArgumentException(nameof(span), "The destination span is too short.");
+        }
+
+        return writtenBytes;
+    }
+
+    /// <summary>
+    /// Tries to get the constant buffer from an input D2D1 pixel shader.
+    /// </summary>
+    /// <typeparam name="T">The type of D2D1 pixel shader to retrieve info for.</typeparam>
+    /// <param name="shader">The input D2D1 pixel shader to retrieve info for.</param>
+    /// <param name="span">The target <see cref="Span{T}"/> to write the constant buffer to.</param>
+    /// <param name="bytesWritten">The number of bytes written into <paramref name="span"/>.</param>
+    /// <returns>Whether or not the constant buffer was retrieved successfully.</returns>
+    public static unsafe bool TryGetConstantBuffer<T>(in T shader, Span<byte> span, out int bytesWritten)
```

Comment:
The behavior here is consistent with how other `Try___` methods work. The maximum byte size of a constant buffer is capped at 64 DWORDs, so 256 bytes, as per HLSL spec, so you can just always allocate a 256 bytes buffer and use that. My generator will just always emit a build error it a constant buffer in a shader was greater than that, so it's guaranteed that that maximum size will always be respected 🙂

Reply:
Which other `Try___` methods? If you're going for similarity with `Span.TryCopyTo()`, the analogy doesn't hold because you've got the spans right there and you can just inspect their `Length`. All Win32 APIs that involve a caller-allocated buffer let you pass in `null` so that you can query the size first.

How am/was I supposed to know that the maximum is 256 bytes? Just use a magic number in my code? How is anyone else supposed to know? This API really is not useful or robust without a way to ask for the necessary buffer size. This is 100% part of what I was asking for and need!

Reply:
I was looking at these `INumber<T>` APIs @tannergooding mentioned on Discord ([APIs here](https://source.dot.net/#System.Private.CoreLib/IBinaryInteger.cs,23540625848a1acf), [Discord message here](https://discord.com/channels/143867839282020352/960223751599976479/984191899420229642)). I'm not entirely sure whether the best approach here would be to have `bytesWritten` return the required byte size in case of failure (I guess we'd need a different name then), or whether we should add a new APIs such as `GetConstantBufferSize()` like `INumber<T>` does as well 🤔

Reply:
Either way is fine by me, but you don't really need a `Try` version of the method if you can query the size ahead of time with another method

Reply:
Added `D2D1PixelShader.GetConstantBufferSize<T>()` as well 🙂